### ISSUE TITLE
BUG: does not work properly with tsconfig module, moduleResolution, target

### ISSUE THREAD

**Issue by alexn-s** [gvergnaud/hotscript#96]
version: `1.0.11`
typescript: `5.0.4`

package.json
```json
{
  "name": "hot",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "type": "module",
  "scripts": {},
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@types/node": "^18.16.0",
    "hotscript": "^1.0.11",
    "typescript": "^5.0.4"
  }
}

```

### does not work
tsconfig.json 
```json
{
  "compilerOptions": {
    "module": "ESNext",
    "moduleResolution": "nodenext",
    "target": "ESNext",
    "declaration": true,
    "esModuleInterop": true,
    "strict": true,
    "outDir": "dist/"
  },
  "include": ["src/"]
}
```

![image](https://user-images.githubusercontent.com/60710873/236611666-7dbd1805-4242-4bdb-8c18-556bd09b3b2b.png)

### does work
tsconfig.json 
```json
{
  "compilerOptions": {
    "declaration": true,
    "esModuleInterop": true,
    "strict": true,
    "outDir": "dist/"
  },
  "include": ["src/"]
}
```

![image](https://user-images.githubusercontent.com/60710873/236611651-fa4cfafc-6f29-439f-b7a1-89ee31eb1edb.png)

### issue
hotscript does not appear to work with these settings
```json
...
    "module": "ESNext",
    "moduleResolution": "nodenext",
    "target": "ESNext",
...
```

does anybody have any ideas?
just wanted to use hotscript in my code but this "bug" currently stops me from using it.

**Comment by gvergnaud** [comment]
Hey! repro'd this issue locally and I managed to fix it in version `1.0.12-rc.0`. @alexn-s could you try installing on your side and check if everything works?

**Comment by alexn-s** [comment]
@gvergnaud yes it works now.

thank you for this quick fix.
cant wait to use hotscript.

it is really innovative and it has a nice & elegant api
